src/server/rdb_save.cc

```diff
@@ -1097,14 +1097,32 @@ void RdbSaver::Impl::Cancel() {
 }
 
 size_t RdbSaver::Impl::GetTotalBuffersSize() const {
-  DCHECK_EQ(shard_snapshots_.size(), 1u) << "Only supported for dragonfly replication";
-  auto& snapshot = shard_snapshots_.front();
-
-  // Calculate number of enqueued bytes as difference between pushed and pulled
-  size_t enqueued_bytes = snapshot->pushed_bytes() - stats_.pulled_bytes;
-  size_t serializer_bytes = snapshot->GetTotalBufferCapacity();
+  std::atomic<size_t> pushed_bytes{0};
+  std::atomic<size_t> serializer_bytes{0};
+  size_t pulled_bytes = stats_.pulled_bytes;
+
+  if (shard_snapshots_.size() == 1) {
+    auto& snapshot = shard_snapshots_.front();
+    pushed_bytes.store(snapshot->pushed_bytes(), memory_order_relaxed);
+    serializer_bytes.store(snapshot->GetTotalBufferCapacity(), memory_order_relaxed);
+  } else {
+    auto cb = [this, &pushed_bytes, &serializer_bytes](ShardId sid) {
+      auto& snapshot = shard_snapshots_[sid];
+      pushed_bytes.fetch_add(snapshot->pushed_bytes(), memory_order_relaxed);
+      serializer_bytes.store(snapshot->GetTotalBufferCapacity(), memory_order_relaxed);
+    };
+    shard_set->RunBriefInParallel([&](EngineShard* es) { cb(es->shard_id()); });
+    // Note that pushed bytes and pulled bytes values are fetched at different times, as we need to
+    // calc the pushed bytes using RunBriefInParallel.
+    // pulled bytes might be higher untill we return here from RunBriefInParallel.
```

Comment:
that's true... this figure is not accurate 🤔 But let's keep it like now, it's the higher bound

Reply:
yes, I will keep it like this, this is only a comment so that it will be clear that this number is not accurate